File: CodeWars/CodeGarden/21-Vasya-Clerk.js
function tickets(peopleInLine){

    let register = {
      25: 0,
      50: 0,
      100: 0
    };
      
    let answer = "No";
    
    for(let i=0; i<peopleInLine.length; i++){
      const payment = peopleInLine[i];
      if(payment === 25) {
        register[payment]++;
      }else if (payment === 50 && register[25] > 0){
        register[payment]++;
        register[25]--;
      }else if (payment === 100){
         if(register[25] > 0 && register[50] > 0){
          register[25]--;
          register[50]--;
          register[payment]++;
        }else if(register[25] > 2){
          register[25] -=3;
          register[payment]++;
        }else{
        return "NO";
      }
      }else{
        return "NO";
      }
    }
    return "YES";
  }

File: CodeWars/CodeGarden/21-Vasya-Clerk.cs
using System;

public class Line
{
    public static string Tickets(int[] peopleInLine)
    {
        int twentyFives = 0, fifties = 0;

        foreach (var bill in peopleInLine)
        {
            switch (bill)
            {
                case 25:
                    ++twentyFives;
                    break;
                case 50:
                    --twentyFives;
                    ++fifties;
                    break;
                case 100:
                    if (fifties == 0)
                    {
                        twentyFives -= 3;
                    }
                    else
                    {
                        --twentyFives;
                        --fifties;
                    }
                    break;
            }

            if (twentyFives < 0 || fifties < 0)
            {
                return "NO";
            }
        }

        return "YES";
    }
}